M2/Macaulay2/m2/testing.m2

```diff
@@ -76,23 +105,23 @@ check(ZZ, Package) := opts -> (n, pkg) -> (
     --
     use pkg;
     if pkg#?"documentation not loaded" then pkg = loadPackage(pkg#"pkgname", LoadDocumentation => true, Reload => true);
-    if not pkg#?"test directory loaded" then loadTestDir pkg;
-    tests := if n == -1 then toList(0 .. pkg#"test number" - 1) else {n};
-    if #tests == 0 then printerr("warning: ", toString pkg,  " has no tests");
+    inputs := tests pkg;
+    testKeys := if n == -1 then toList(0 .. pkg#"test number" - 1) else {n};
+    if #testKeys == 0 then printerr("warning: ", toString pkg,  " has no tests");
     --
     errorList := {};
     (hadError, numErrors) = (false, 0);
-    scan(tests, k -> (
-	    (filename, lineno, teststring) := pkg#"test inputs"#k;
+    scan(testKeys, k -> (
+	    if not inputs#?k then error(pkg, " has no test #", k);
+	    teststring := code inputs#k;
 	    desc := "check(" | toString k | ", " | format pkg#"pkgname" | ")";
 	    ret := elapsedTime captureTestResult(desc, teststring, pkg, usermode);
 	    if not ret then errorList = append(errorList,
 		 (k, temporaryFilenameCounter - 2))));
     outfile := k -> temporaryDirectory() | toString k | ".tmp";
     if hadError then (
 	if opts.Verbose then apply(errorList, (j, k) -> (
-		(filename, lineno, teststring) := pkg#"test inputs"#j;
-		stderr << filename << ":" << lineno - 1 << ":1: error:" << endl;
+		stderr << toString inputs#k << " error:" << endl;
```

Comment:
I introduced a bug here -- this should be `inputs#j`.  It's fixed locally -- I'll push it along with some other improvements soon.